src/pd/client.rs

```diff
@@ -30,31 +30,95 @@ pub trait TRpcClient: Sync + Send {
 }
 
 #[derive(Debug)]
+struct RpcClientCore {
+    addrs: Vec<String>,
+    next_index: usize,
+    stream: Option<TcpStream>,
+}
+
+fn connect(addr: &str) -> Result<TcpStream> {
+    let stream = try!(make_std_tcp_conn(addr));
+    try!(stream.set_read_timeout(Some(Duration::from_secs(SOCKET_READ_TIMEOUT))));
+    try!(stream.set_write_timeout(Some(Duration::from_secs(SOCKET_WRITE_TIMEOUT))));
+    Ok(stream)
+}
+
+impl RpcClientCore {
+    fn new(dsn: &str) -> RpcClientCore {
+        let addrs: Vec<String> = dsn.split(',').map(String::from).collect();
+        RpcClientCore {
+            addrs: addrs,
+            next_index: 0,
+            stream: None,
+        }
+    }
+
+    fn try_connect(&mut self) -> Result<()> {
+        let index = self.next_index;
+        self.next_index += 1;
+        if self.next_index >= self.addrs.len() {
```

Comment:
Use self.next_index = self.next_index % self.addrs.len() instead of these three lines.